### DISCUSSION TITLE
Proposal for detecting correct ChargingScheduleAllowedChargingRateUnit

### DISCUSSION THREAD

**Discussion by rbb-dev** [lbbrhzn/ocpp#1322] [Ideas]
Hi,

There are chargers (i'm lucky to have those) that do not conform to the standard and don't specify Current or Power in the ChargingScheduleAllowedChargingRateUnit config. (see below),

{"key":"ChargingScheduleAllowedChargingRateUnit","readonly":true,"value":"(NULL)"}
{"key":"ChargingScheduleAllowedChargingRateUnit","readonly":true,"value":""}

One only works in amps and the other in watts

Can I propose that detention be made by sending txdefault with a startPeriod of some time in the future (to avoid immediate charging session, or   and testing for accept/reject on different settings of chargingRateUnit?
After detection, charge profiles can be cleared or tx default replaced.
If Current/Power is set, there is nothing to do, but detection is kicked in if values are missing or invalid, as in the examples above.

Any thoughts?

**Comment by drc38** [comment]
Best to simply use the `set_charge_rate` action and a custom slider or similar with automation.

**Reply by drc38** [reply]
You'll need to send it a new profile using an automation each time you want to alter the charging limit.
